docs/source/developer-guidelines/images.md

```diff
@@ -0,0 +1,62 @@
+# Images
+
+The goals for optimizing images are:
+
+- serving proper size images to the client (using `srcset`s)
+- lazy loading images
+
+The most basic example of image usage is an image block, or an image in Tinymce if using Classic UI.
+In both situations the user usually defines an alignment and a size for the image.
+
+The alignment can be: left, centered, right, or full width.
+
+The size can be: small, medium, large.
+Depending on the site layout and design, these sizes can be fixed (in pixels) or fluid (in percentages).
+
+## Serving the right image size
+
+In order to achieve the first goal, images have to be rendered using a `picture` element and a `source` element with a `srcset` attribute.
+
+```html
+<picture>
+  <source
+    srcset="
+      /image-400-uuid.jpg   400w,
+      /image-600-uuid.jpg   600w,
+      /image-800-uuid.jpg   800w,
+      /image-1000-uuid.jpg 1000w,
+      /image-1200-uuid.jpg 1200w
+    "
+    sizes="100vw"
+  />
+  <img
+    src="/image-placeholder.jpg"
+    alt="Alt text"
+    role="img"
+    width="960"
+    height="600"
+  />
+</picture>
+```
+
+This will allow the browser to choose the proper size to download from the server.
+
+### Width and height
+
+The important thing to note here is that we provide width and height in the `img` tag, those are used by the browser to define which space the image will take and avoid layout shift while loading the page. These information are to be taken from the backend. The backend will provide urls for each size for the `srcset` and the width and height attribute for each one.
+
+The size of the image chosen by the user defines which of the combinations of width and height are used among those served by the backend.
+
+### Other discussion points
+
+_Should we render in the `srcset` the sizes that are bigger than the chosen one?_
+
+_In fluid layouts, the alignment decision made by the user will impact the actual width of the image, based on CSS rules. Should this be taken into account for `srcset` rendering?_
+
+Sizes are defined in the Volto config, or in the Tinymce controlpanel if using Classic UI.
+
+## Lazy loading
```

Comment:
Nowadays, lazy loading image is well implemented by the `loading="lazy"` attribute which is widely supported by all browsers https://caniuse.com/loading-lazy-attr (pay attention that for some browser lazy attribute is not supported for iframes, but it is for images). 

IMHO adding `IntersetionObserver` may not be necessary or have a negative impact.

Reply:
Only the very latest versions of Safari support this attribute by default: https://caniuse.com/?search=loading%20lazy
This feature has an overall 80% support globally among browsers, while [IntersectionObserver is way higher with 94%](https://caniuse.com/?search=intersectionobserver).

In addition to this, I'm not sure if we want to lazy load the tiny resolution placeholder that we use before the srcset is applied by the IntersectionObserver, more thought needs to be given on this decision imho.

Moreover, @MrTango was reporting an issue in Chrome support for `loading="lazy"`, regarding the pixels threshold that the browser tracks for loading the image lazily. @MrTango if you could paste a pointer to the issue it would be great, unless it has been solved already.